rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -0,0 +1,321 @@
+                    event.amount == shares_amount and
+                    event.asset == vault_token
+                ):
+                    event.counterparty = CPT_SPARK
+                    event.event_type = HistoryEventType.RECEIVE
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.notes = f'Receive {shares_amount} {vault_token.symbol} from depositing into Spark Savings'  # noqa: E501
+                    in_event = event
+            elif (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == shares_amount and
+                event.asset == vault_token
+            ):
+                event.counterparty = CPT_SPARK
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {shares_amount} {vault_token.symbol} to Spark Savings'
+                out_event = event
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == underlying_amount and
+                event.asset == underlying_token
+            ):
+                event.counterparty = CPT_SPARK
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Remove {underlying_amount} {underlying_token.symbol} from Spark Savings'  # noqa: E501
+                in_event = event
+
+        action_items = []
+        if context.tx_log.topics[0] == DEPOSIT_TOPIC and in_event is None:
+            # For deposits, create action item for the receive event that's not being decoded
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.RECEIVE,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=vault_token,
+                amount=shares_amount,
+                to_event_type=HistoryEventType.RECEIVE,
+                to_event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+                to_notes=f'Receive {shares_amount} {vault_token.symbol} from depositing into Spark Savings',  # noqa: E501
+                to_counterparty=CPT_SPARK,
+                paired_events_data=([out_event] if out_event else [], True),
+            ))
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
+        return DecodingOutput(action_items=action_items)
+
+    @staticmethod
+    def _decode_sky_migration_to_susds(context: DecoderContext) -> DecodingOutput:
+        """Decodes Sky protocol migrations from sDAI/DAI to sUSDS tokens.
+        It is here to avoid decoder mapping conflicts. Only processes Ethereum migrations.
+        """
```

Comment:
this looks like it belong to the SKY decoder. Can you explain again why you had to move it here?